src/utils.rs

```diff
@@ -6,3 +9,18 @@ pub fn set_threading(num_threads: usize) {
         .build_global()
         .unwrap();
 }
+
+cached_key! {
+    READ: SizedCache<String, Array<f64, IxDyn>> = SizedCache::with_size(10);
+    Key = { format!("{}", path)};
+    fn read_nd_image(path: String) -> Array<f64, IxDyn> = {
+        let nifti_object = InMemNiftiObject::from_file(path).expect("Nifti file is unreadable.");
+        let volume = nifti_object.into_volume();
+        volume.into_ndarray::<f64>().unwrap()
+    }
+}
+
+pub fn read_header(path: &str) -> NiftiHeader {
+    let nifti_object = InMemNiftiObject::from_file(path).expect("Nifti file in unreadable.");
+    nifti_object.header().clone()
```

Comment:
You need to know that this function reads the whole image anyway :) I'm not sure if there something in nifti-rs to actually read only the header. And you don't need the `.clone()`.

Reply:
Yes there is, I'll use it https://github.com/Enet4/nifti-rs/blob/master/src/header.rs#L557.